shapely/tests/test_strtree.py

```diff
@@ -65,151 +68,286 @@ def poly_tree():
     ],
 )
 def test_init(geometry, count, hits):
-    tree = shapely.STRtree(np.array(geometry))
+    tree = STRtree(geometry)
     assert len(tree) == count
-    assert tree.query_items(box(0, 0, 100, 100)).size == hits
+    assert tree.query(box(0, 0, 100, 100)).size == hits
 
 
 def test_init_with_invalid_geometry():
     with pytest.raises(TypeError):
-        shapely.STRtree(np.array(["Not a geometry"], dtype=object))
+        STRtree(["Not a geometry"])
 
 
 def test_init_increases_refcount():
     arr = np.array([point])
     with assert_increases_refcount(point):
-        _ = shapely.STRtree(arr)
+        _ = STRtree(arr)
 
 
 def test_del_decreases_refcount():
     arr = np.array([point])
-    tree = shapely.STRtree(arr)
+    tree = STRtree(arr)
     with assert_decreases_refcount(point):
         del tree
 
 
+def test_references():
+    point1 = Point()
+    point2 = Point(0, 1)
+
+    geoms = [point1, point2]
+    tree = STRtree(geoms)
+
+    point1 = None
+    point2 = None
+
+    import gc
+
+    gc.collect()
+
+    # query after freeing geometries does not lead to segfault
+    assert tree.query(box(0, 0, 1, 1)).tolist() == [1]
+
+
 def test_flush_geometries():
     arr = shapely.points(np.arange(10), np.arange(10))
-    tree = shapely.STRtree(arr)
+    tree = STRtree(arr)
+
     # Dereference geometries
+    tree.geometries.flags.writeable = True
     arr[:] = None
     import gc
 
     gc.collect()
     # Still it does not lead to a segfault
-    tree.query_items(point)
+    tree.query(point)
 
 
 def test_geometries_property():
     arr = np.array([point])
-    tree = shapely.STRtree(arr)
+    tree = STRtree(arr)
     assert arr is tree.geometries
 
 
-def test_query_invalid_geometry(tree):
-    with pytest.raises(TypeError):
-        tree.query_items("I am not a geometry")
+# TODO(shapely-2.0) this fails on Appveyor, see
+# https://github.com/shapely/shapely/pull/983#issuecomment-718557666
+@pytest.mark.skipif(sys.platform.startswith("win32"), reason="does not run on Appveyor")
```

Comment:
This supersedes `tests/test_strtree.py::test_pickle_persistence` and instead of using a dedicated test script `unpickle-strtree.py` it writes it out temporarily.  This will make it easier to maintain alongside the tests.  The logic in the unpickle script is the same as before, except for removing unneeded print statements and comments.